src/ORM/FieldType/DBString.php

```diff
@@ -49,10 +54,13 @@ public function setOptions(array $options = []): static
     {
         parent::setOptions($options);
 
-        if (array_key_exists('nullifyEmpty', $options ?? [])) {
+        if (array_key_exists('nullifyEmpty', $options)) {
             $this->options['nullifyEmpty'] = (bool) $options['nullifyEmpty'];
         }
-        if (array_key_exists('default', $options ?? [])) {
+        if (array_key_exists('default', $options)) {
+            if ($this->getNullifyEmpty() && $this->isEmptyValue($options['default'])) {
+                $options['default'] = null;
+            }
```

Comment:
If we're nullifying empty, the empty string default should be null. Otherwise we're _not_ nullifying empty.

Another solution would be to pass the default through `prepValueForDB()` (either in `setDefaultValue()` or in the various `getFieldSpec()` implementations) - happy to go that route if you prefer.